lib/sneakers/handlers/maxretry.rb

```diff
@@ -165,9 +165,14 @@ def failure_count(headers)
         if headers.nil? || headers['x-death'].nil?
           0
         else
-          headers['x-death'].select do |x_death|
+          x_death_array = headers['x-death'].select do |x_death|
             x_death['queue'] == @worker_queue_name
-          end.count
+          end
+          if x_death_array.count != 1
+            x_death_array.count
+          else
+            x_death_array.first['count'] || 1
```

Comment:
If I understand, for the "retired once case", using an older version of rabbit, there will be no `count` attribute, and the above `x_death_array` count will be 1, so `x_death_array.first['count']` will return nil, and so you'll return 1, does that sound about right?

Also, what happens when `x_death_array` is 0? Won't this raise an exception? Should there be some protection if that's the case (although it might not be something Rabbit would intentionally do).